_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,208 @@
+
+array.dask  # shows the HTML representation in Jupyter
+```
+
+<img src="/images/gsoc21/7886.png" alt="Example: Array images now included in HTML representation of Dask high level graphs, pull request #7886" height=414 width=736>
+
+Reference: [Pull request #7886 by Freyam Mehta *"Add dask.array SVG to the HTML Repr"*](https://github.com/dask/dask/pull/7886)
+
+### New HTML repr for ProcessInterface class
+
+A new HTML representation has been created for the `ProcessInterface` class in [dask distributed](https://github.com/dask/distributed/). The `ProcessInterface` class is not intended to be used directly. Instead, this information should be accessed via SSH scheduler or workers.
+
+The HTML representation displays the status, address, and external address of the process.
+
+There are three possible status options:
+* Process created, not yet running (blue icon)
+* Process is running (green icon)
+* Process closed (orange icon)
+
+Example:
+
+```python
+from dask.distributed import LocalCluster, Client, SSHCluster
+
+cluster = SSHCluster(["127.0.0.1", "127.0.0.1", "127.0.0.1"])
+cluster.scheduler  # HTML representation for the SSH scheduler, shown in Jupyter
+# or
+cluster.workers[0]  # HTML representation for the first SSH worker in the cluster
+```
+
+<img src="/images/gsoc21/5181.png" alt="Example: New HTML representation for distributed ProcessInterface class, pull request #5181" height=414 width=736>
+
+Reference: [Pull request #5181 by Freyam Mehta *"Add HTML Repr for ProcessInterface Class and all its subclasses"*](https://github.com/dask/distributed/pull/5181)
+
+### New HTML repr for Security class
+
+Pull request [#5178](https://github.com/dask/distributed/pull/5178) added a new HTML representation for the `Security` class in the [dask distributed library](https://github.com/dask/distributed/). In addition, the text representation has also been updated to reflect the same information shown in the HTML representation.
+
+
+The `Security` HTML representation shows:
+* Whether encryption is required
+* Whether the object instance was created using `Security.temporary()` or `Security(**paths_to_keys)`.
+    * For temporary security objects, keys are generated dynamically and the only copy is kept in memory.
+    * For security objects created using keys stored on disk, the HTML representation will show the full filepath to the relevant security certificates on disk.
+
+
+Example: temporary security object
+
+```python
+from dsak.distributed import Security
+
+s = Security.temporary()
+s  # shows the HTML representation in Jupyter
+```
+
+Example: security object using certificates saved to disk
+
+```python
+from dsak.distributed import Security
+
+s = Security(require_encryption=True, tls_ca_file="ca.pem", tls_scheduler_cert="scert.pem")
+s  # shows the HTML representation in Jupyter
+```
+
+<img src="/images/gsoc21/5178.png" alt="Example: New HTML representation for distributed Security class, pull request #5178" height=414 width=736>
```

Comment:
@freyam the image demo for the Security HTML repr needs to be added to the `images` folder.